### PR TITLE
37 bonus question

### LINKED ISSUE #37: Bonus Question Screen
### Description

Add a bonus question screen with an interactive "puzzle" where the user can double his/her points if it's answered correctly. Should he/she get the answer wrong, then the player also loses the points he/she received in on the previous screen.

### Tasks

- [x] The screen displays clickable nodes. Each node has a corresponding movie/series poster attached.
- [x] The player needs to decide whether the already correctly guessed movie was released before, between or after the displayed movies.
- [x] If the player answers correctly, the [Success Screen](https://github.com/kwizapp/kwiz/issues/12) is shown and points are added (but no bonus question is possible from there).
- [x] If the player answers incorrectly, the [Failure Screen](https://github.com/kwizapp/kwiz/issues/13) is shown and points are deducted.

### Context 
[Story: Question Screen ("Bonus")](https://github.com/kwizapp/kwiz/issues/15)

### Additional Ideas
- Fruit-Ninja mode, where the player needs to slice up wrong facts of a move (the facts rain down on the screen)

### PR DESCRIPTION
**Pull request by rschlaefli**
### Description

Implement a basic bonus question: the correctly guessed movie is shown and needs to be inserted into a release timeline containing the other three movies (the same ones as in the preceding question).

image

As the failure screen is currently built specifically for the normal question screen, it does not show the correct reason for failing (i.e., "you chose the wrong year bracket"). I have created a separate issue (https://github.com/kwizapp/ionic-app/issues/103) for this and have not addressed it here.

### Related Issues

- Closes #37 
- https://github.com/kwizapp/kwiz/issues/15
- Creates https://github.com/kwizapp/ionic-app/issues/103

### Checklist

- [x] Have you added tests where necessary? Do all the test pass? 
- [x] Have you added descriptive comments to your code?
- [x] Have you updated the documentation related to this propo

### CODE REVIEW

File: src/components/card/MovieCard.tsx [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 
 interface Props {
   title: string
-  posterPath: string
-  onClick: (title: string) => void
+  posterPath?: string
+  onClick?: (title: string) => void
 }
 
 const MovieCard = ({ title, posterPath, onClick }: Props) => {
```
Comment (rschlaefli) [inline src/components/card/MovieCard.tsx:10]: These changes make the `MovieCard` more generic and usable on the bonus screen as well

File: src/pages/BonusQuestion.tsx [hunk 2]
```diff
@@ -1,34 +1,90 @@
...
+`
+
 const BonusQuestion = () => {
   const history = useHistory()
 
-  const { setPointDifference } = useStore()
+  const { setPointDifference, pointDifference } = useStore()
+
+  const { loading, data } = useQuery<MovieData>(MOVIES, {
+    fetchPolicy: 'cache-only',
```
Comment (rschlaefli) [inline src/pages/BonusQuestion.tsx:27]: Use the same random movies as in the initial question. We could also put
```suggestion
    fetchPolicy: 'network-only',
```
if we want three different random movies.

File: src/pages/Success.tsx [hunk 3]
```diff
@@ -69,20 +69,17 @@ const Success = () => {
   // specify order of animations, will be animated one after the other
   useChain([emojiAnimRef, appearAnimRef, pointsAnimRef, panelAnimRef])
 
-  const updateScores = () => {
+  const goToTriviaScreen = () => {
     // set the points here, as we wanted to animate the difference
     addPoints(pointDifference)
-    // reset point difference for next round
+    // reset point difference for next round. we only do that when we go to the trivia screen
```
Comment (rschlaefli) [inline src/pages/Success.tsx:75]: Important to keep around the points won in the normal question, such that they can be doubled in the bonus round (or lost!)

**Review by alexscheitlin — APPROVED** [review]
Nice, thanks. Let's merge this.
In a second step, we could visually "separate" the movie from the first question (e.g., "47 Meters Down").

**Review by nikzaugg — APPROVED** [review]
Looks Good.
I am merging this.

And I will open another issue to polish some of the inconsistencies :+1:

**Pull request merged**